src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/CertCollectionLoader.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+
+namespace Internal.Cryptography.Pal
+{
+    internal class CertCollectionLoader : ILoaderPal
```

Comment:
From my perspective there are three main reasons to mark a type as sealed:
1. Prevent unexpected usage from others deriving from the type.
2. Enable better performance by allowing for better devirtualization by the backend compiler.
3. Enable better code understandability; you look at a non-null variable of type X, and if it's sealed, you know that the instance referenced by that variable is definitively of that type and not something derived from it that may be overriding functionality, re-implementing an interface, adding more fields to make the instance more heavyweight, etc.

For an internal type, (1) isn't so much of an issue, and theoretically a compiler could do (2), but regardless (3) still applies.